libs/datamodel/schema-ast/src/ast/find_at_position.rs

```diff
@@ -53,6 +53,8 @@ pub enum ModelPosition<'ast> {
     ModelAttribute(&'ast str, usize),
     /// In a field.
     Field(ast::FieldId, FieldPosition<'ast>),
+    /// In an index attribute.
+    Index(usize, AttributePosition<'ast>),
```

Comment:
Why not use `ModelAttribute` here? My preference would be to either always specialize per attribute type (but imo this is not good for separation of concerns — some attributes are connector-specific) or use the general `ModelAttribute`. But feel free to ignore.

Reply:
I'll try that... It's hard to get immediately the separation of concerns here.

Reply:
yep it's not super well structured, open to anything

Reply:
my intuition is that if you do a match on a position, you can match on `ModelPosition::ModelAttribute("index", AttributePosition(...))`